components/org.wso2.carbon.siddhi.extensions.installer.core/src/main/java/org/wso2/carbon/siddhi/extensions/installer/core/execution/DependencyInstallerImpl.java

```diff
@@ -0,0 +1,271 @@
+                        LOGGER.debug(String.format("Failed to uninstall dependency: %s.",
+                            dependency.getRepresentableName()));
+                    }
+                }
+            }
+            ExtensionUnInstallationStatus status =
+                ExtensionsInstallerUtils.getUnInstallationStatus(
+                    completedDependencies.size(), extension.getDependencies().size());
+            return ResponseEntityCreator.createExtensionUnInstallationResponse(
+                status, completedDependencies, failedDependencies);
+        }
+        throw new ExtensionsInstallerException(String.format(
+            "No dependencies were specified in extension: %s", extensionId));
+    }
+
+    private boolean unInstallUsagesFor(DependencyConfig dependency) throws ExtensionsInstallerException {
+        String lookupRegex = dependency.getLookupRegex();
+        if (lookupRegex != null) {
+            boolean hasFailures = false;
+            for (UsageConfig usage : dependency.getUsages()) {
+                // Delete jar(s) for the usage from the final directory.
+                boolean deletedInFinalDirectory =
+                    deleteMatchingJars(lookupRegex, ExtensionsInstallerUtils.getBundleLocation(usage));
+                hasFailures = !deletedInFinalDirectory;
+
+                // Only for non-bundle usages, delete jar(s) from the initial directory as well.
+                if (usage.getType() == UsageType.JAR) {
+                    boolean deletedInInitialDirectory =
+                        deleteMatchingJars(lookupRegex, ExtensionsInstallerUtils.getInstallationLocation(usage));
+                    hasFailures = hasFailures || !deletedInInitialDirectory;
+                }
+            }
+            return !hasFailures;
+        } else {
+            throw new ExtensionsInstallerException("Unable to find property: 'lookupRegex'.");
+        }
+    }
+
+    private boolean deleteMatchingJars(String lookupRegex, String directoryPath) {
+        boolean allJarsDeleted = false;
+        try {
+            /*
+			Ideally there will be just one matching jar,
+			unless a jar with a different version (but with the same name) is manually placed in the directory.
+			*/
+            List<Path> versionedJarFiles = ExtensionsInstallerUtils.listMatchingFiles(lookupRegex, directoryPath);
+            if (!versionedJarFiles.isEmpty()) {
+                allJarsDeleted = deleteJarFiles(versionedJarFiles);
+            }
+        } catch (IOException e) {
+            LOGGER.error(String.format("Failed to list matching files for lookup regex: %s.", lookupRegex));
+        }
+        return allJarsDeleted;
+    }
+
+    private boolean deleteJarFiles(List<Path> jarFiles) {
+        boolean hasFailures = false;
+        for (Path jarFile : jarFiles) {
+            try {
+                Files.deleteIfExists(jarFile);
+            } catch (IOException e) {
+				/*
```

Comment:
Format the comment

Reply:
Addressed in https://github.com/wso2/carbon-analytics/pull/1859